src/richie/apps/core/context_processors.py

```diff
@@ -86,3 +92,92 @@ def site_metas(request):
 
     context["FRONTEND_CONTEXT"] = json.dumps(context["FRONTEND_CONTEXT"])
     return context
+
+
+class WebAnalyticsContextProcessor:
+    """
+    Context processor to add Web Analytics tracking information to Richie CMS templates and
+    frontend.
+    """
+
+    def context_processor(self, request: HttpRequest) -> dict:
+        """
+        Real implementation of the context processor for the Web Analytics core app sub-module
+        """
+        context = {}
+        if hasattr(request, "current_page"):
+            context["WEB_ANALYTICS_GOOGLE_ANALYTICS_ID"] = getattr(
+                settings, "WEB_ANALYTICS_GOOGLE_ANALYTICS_ID", None
+            )
+            context["WEB_ANALYTICS_LOCATION"] = getattr(
+                settings, "WEB_ANALYTICS_LOCATION", None
+            )
+            if self.is_load_dimensions(request):
+                context["WEB_ANALYTICS_DIMENSIONS"] = self.get_dimensions(request)
+        return context
+
+    # pylint: disable=no-self-use
+    def is_load_dimensions(self, request: HttpRequest) -> bool:
+        """
+        Method to easily enable or disable the load of web analytics dimensions
+        """
+        return getattr(settings, "WEB_ANALYTICS_LOCATION", None)
```

Comment:
I think this is a typo as the location setting is something else? you're probably missing a test to secure this case. I didn't understand why we sometimes wouldn't want to get the dimensions?

Reply:
My ideia is not to include the dimensions if the app won't have the web analytics active. Currently the PR needs the `WEB_ANALYTICS_LOCATION` defined, but I think it's better if we include the `WEB_ANALYTICS_PROVIDER` setting (view previous conversation)